syntax = "proto3";

package iota.rpc;
import "proto/messages.proto";

service Hub {
  rpc CreateUser              (CreateUserRequest             ) returns (       CreateUserReply       );
  rpc GetBalance              (GetBalanceRequest             ) returns (       GetBalanceReply       );
  rpc GetGetDepositAddress    (GetDepositAddressRequest      ) returns (       GetDepositAddressReply);
  rpc ScheduleSweep           (ScheduleSweepRequest          ) returns (       ScheduleSweepReply    );
  rpc UserWithdraw            (UserWithdrawRequest           ) returns (       UserWithdrawReply     );
  rpc GetStats                (StatsRequest                  ) returns (       StatsReply            );
  rpc GetUserHistory          (GetUserHistoryRequest         ) returns (       GetUserHistoryReply   );
  rpc UserBalanceSubscription (UserBalanceSubscriptionRequest) returns (stream UserBalanceEvent      );
}

